subprojects/dependency-management/src/main/java/org/gradle/api/internal/artifacts/ivyservice/ivyresolve/RepositoryChainDependencyToComponentIdResolver.java

```diff
@@ -58,7 +58,7 @@ public void resolve(DependencyMetadata dependency, BuildableComponentIdResolveRe
             ResolvedVersionConstraint resolvedVersionConstraint = new DefaultResolvedVersionConstraint(raw, versionSelectorScheme);
             VersionSelector preferredSelector = resolvedVersionConstraint.getPreferredSelector();
             if (preferredSelector.isDynamic()) {
-                dynamicRevisionResolver.resolve(toModuleDependencyMetadata(dependency), preferredSelector, result);
+                dynamicRevisionResolver.resolve(toModuleDependencyMetadata(dependency), preferredSelector, resolvedVersionConstraint.getRejectedSelector(), result);
```

Comment:
I am just curious: Is this a bugfix? Or why did it work for `strictly()` without this?

Reply:
It wasn't needed before, because we computed the rejection clause so we knew it didn't overlap with the prefer version. This isn't true anymore.